deye_sensors.py

```diff
@@ -47,39 +47,48 @@
 pv1_current_sensor = SingleRegisterSensor("PV1 Current", 0x6e, 0.1, mqtt_topic_suffix='dc/pv1/current')
 pv1_power_sensor = ComputedPowerSensor("PV1 Power", pv1_voltage_sensor,
                                        pv1_current_sensor, mqtt_topic_suffix='dc/pv1/power')
+pv1_daily_sensor = SingleRegisterSensor("PV1 Daily", 0x41, 0.1, mqtt_topic_suffix='dc/pv1/daily')
+pv1_total_sensor = DoubleRegisterSensor("PV1 Total", 0x45, 0.1, mqtt_topic_suffix='dc/pv1/total')
 
 # DC PV2
 pv2_voltage_sensor = SingleRegisterSensor("PV2 Voltage", 0x6f, 0.1, mqtt_topic_suffix='dc/pv2/voltage')
 pv2_current_sensor = SingleRegisterSensor("PV2 Current", 0x70, 0.1, mqtt_topic_suffix='dc/pv2/current')
 pv2_power_sensor = ComputedPowerSensor("PV2 Power", pv2_voltage_sensor,
                                        pv2_current_sensor, mqtt_topic_suffix='dc/pv2/power')
+pv2_daily_sensor = SingleRegisterSensor("PV2 Daily", 0x42, 0.1, mqtt_topic_suffix='dc/pv2/daily')
+pv2_total_sensor = DoubleRegisterSensor("PV2 Total", 0x47, 0.1, mqtt_topic_suffix='dc/pv2/total')
 
 # DC PV3
 pv3_voltage_sensor = SingleRegisterSensor("PV3 Voltage", 0x71, 0.1, mqtt_topic_suffix='dc/pv3/voltage')
 pv3_current_sensor = SingleRegisterSensor("PV3 Current", 0x72, 0.1, mqtt_topic_suffix='dc/pv3/current')
 pv3_power_sensor = ComputedPowerSensor("PV3 Power", pv3_voltage_sensor,
                                        pv3_current_sensor, mqtt_topic_suffix='dc/pv3/power')
+pv3_daily_sensor = SingleRegisterSensor("PV3 Daily", 0x43, 0.1, mqtt_topic_suffix='dc/pv3/daily')
 
 # DC PV4
 pv4_voltage_sensor = SingleRegisterSensor("PV4 Voltage", 0x73, 0.1, mqtt_topic_suffix='dc/pv4/voltage')
 pv4_current_sensor = SingleRegisterSensor("PV4 Current", 0x74, 0.1, mqtt_topic_suffix='dc/pv4/current')
 pv4_power_sensor = ComputedPowerSensor("PV4 Power", pv4_voltage_sensor,
                                        pv4_current_sensor, mqtt_topic_suffix='dc/pv4/power')
+pv4_daily_sensor = SingleRegisterSensor("PV4 Daily", 0x44, 0.1, mqtt_topic_suffix='dc/pv4/daily')
 
 # Power sensors
 operating_power_sensor = SingleRegisterSensor("Operating Power", 0x50, 0.1, mqtt_topic_suffix='operating_power')
 dc_power_sensor = SingleRegisterSensor("DC Total Power", 0x52, 0.1, mqtt_topic_suffix='dc/total_power')
 ac_apparent_power_sensor = SingleRegisterSensor("AC Apparent Power", 0x54, 0.1, mqtt_topic_suffix='ac/apparent_power')
-ac_active_power_sensor = SingleRegisterSensor("AC Active Power", 0x56, 0.1, mqtt_topic_suffix='ac/active_power')
+#ac_active_power_sensor = SingleRegisterSensor("AC Active Power", 0x56, 0.1, mqtt_topic_suffix='ac/active_power')
+ac_active_power_sensor = DoubleRegisterSensor("AC Active Power", 0x56, 0.1, mqtt_topic_suffix='ac/active_power')
 ac_reactive_power_sensor = SingleRegisterSensor("AC Reactive Power", 0x58, 0.1, mqtt_topic_suffix='ac/reactive_power')
+production_total_sensor = DoubleRegisterSensor("Production Total", 0x3f, 0.1, mqtt_topic_suffix='total_energy')
```

Comment:
I confirm, this works well with my Deye :+1:

Reply:
i do not expect that i (or my deye) will ever reach the higher word before end of livetime